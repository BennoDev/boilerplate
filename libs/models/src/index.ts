export * from './models.module';
export * from './entities';
export * from './repositories';
