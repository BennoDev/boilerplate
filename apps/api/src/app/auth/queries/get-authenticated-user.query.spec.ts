import { Test, type TestingModule } from '@nestjs/testing';

import { ModelsModule } from '@libs/models';

import { GetAuthenticatedUserHandler } from './get-authenticated-user.query';

describe('GetAuthenticatedUserHandler', () => {
    let module: TestingModule;
    let handler: GetAuthenticatedUserHandler;

    beforeAll(async () => {
        module = await Test.createTestingModule({
            imports: [ModelsModule.registerTest()],
            providers: [GetAuthenticatedUserHandler],
        }).compile();

        handler = module.get(GetAuthenticatedUserHandler);
    });

    afterAll(async () => {
        await module.close();
    });

    describe('execute', () => {
        it('should return the correct user', async () => {
            const userId = 'c4cb4582-1e97-4e3e-9d49-c744c8c1a229';
            const result = await handler.execute({ data: { userId } });
            expect(result).toMatchSnapshot();
        });
    });
});
