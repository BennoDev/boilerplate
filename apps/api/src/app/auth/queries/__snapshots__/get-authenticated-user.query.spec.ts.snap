// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetAuthenticatedUserHandler execute should return the correct user 1`] = `
{
  "createdAt": "2020-04-17 12:03:42.124+00",
  "email": "development@boilerplate.com",
  "firstName": "Developer",
  "id": "c4cb4582-1e97-4e3e-9d49-c744c8c1a229",
  "lastName": "McDeveloperson",
  "state": "ACTIVE",
  "updatedAt": "2020-04-17 12:03:42.124+00",
}
`;
